cmd/deployment/search.go

```diff
@@ -51,21 +54,57 @@ var searchCmd = &cobra.Command{
 			return err
 		}
 
-		res, err := deploymentapi.Search(deploymentapi.SearchParams{
-			API:     ecctl.Get().API,
-			Request: &sr,
-		})
+		returnAllMatches, _ := strconv.ParseBool(cmd.Flag("all-matches").Value.String())
+		if returnAllMatches && sr.Sort == nil {
+			return fmt.Errorf("The query must include a sort-field when using --all-matches. Example: \"sort\": [\"id\"]")
+		}
 
-		if err != nil {
-			return err
+		var result *models.DeploymentsSearchResponse
+		var cursor string
+		for {
```

Comment:
This is probably just me being overly paranoid, but does it make sense to add some kind of failsafe that kills the loop after a certain (high) number of requests? 

That way a future bug in the AC cursor logic or this tool doesn't end up in an infinite tight loop making search requests?

Reply:
Yeah it could make sense to have some timeout/upper limit